src/inc/volatile.h

```diff
@@ -120,7 +120,7 @@ T VolatileLoad(T const * pt)
     STATIC_CONTRACT_SUPPORTS_DAC_HOST_ONLY;
 
 #ifndef DACCESS_COMPILE
-#if defined(_ARM64_) && defined(__GNUC__)
+#if defined(_ARM64_) && defined(__GNUC__) && !(defined(__BIONIC__))
```

Comment:
The cross build failed regardless of where `__has_builtin(__atomic_load)` was placed with the original error from issue #12982 
```
volatile.h:128:47: error: cannot initialize a parameter of type 'int *' with an rvalue of type 'volatile int *'
```